composer/datasets/in_context_learning_evaluation.py

```diff
@@ -248,10 +263,15 @@ def collate_fn(self, data):
             inputs.append(inp)
             answers.append(aliases)
 
+            # we will search for the answer within the portion of the model response
+            # beginning with `cot_delimiter`
+            cot_delimiter = sample['cot_delimiter']
```

Comment:
Is there always a delimiter? I feel like the options I have seen for CoT are:
* Delimiter
* Answer at end
* Need to evaluate code to get answer (when the model is encouraged to write code to solve the problem, as in Program of Thought in MAmmoTH)

I might be wrong here, maybe the data is presented as if there is no delimiter in the paper, but then they add on `### So the answer is` or something. But if not, then,  maybe we want to have some syntax to support answer-at-end?

Reply:
In Wei's introduction of CoT they use delimiters. I haven't seen any papers in particular that instead check for answer at the end. Our models tend to ramble on and on, so I suspect using delimiters and then checking that the answer is at the beginning of the substring following the delimiter will yield much higher accuracy.

I don't see a very strong need to support both at this time but I could be convinced...

Reply:
Yeah that sounds reasonable to me. Just wanted to talk it out, thank you.